umbraco-deploy/troubleshooting.md

```diff
@@ -10,3 +10,67 @@ If you are unable to find the issue you are having, then please reach out to our
+
+If a lot of files need to checked, this can be slow, and a faster option is available that uses the file metadata. The only downside of changing this option is a marginally increased the chance of Deploy considering a media file hasn't changed when it has.  This would omit it from the deployment.
+
+This option can be [set in configuration](./deploy-setings#mediafilechecksumcalculationmethod).
+
+### Review relation types included in deploy operations
+
+As well as transferring entities between environments Deploy will also include the relations between them. As of 10.1.2 and 11.0.1, two relation types used for usage tracking are omitted by default. These do not need to be transferred as they are recreated by the CMS as part of the save operation on the entity.
+
+If using an earlier version, or to make further adjustements, modify the [settings for relation types](./deploy-setings#relationtypes) in configuration.
+
+## Path too long exceptions
+
+When restoring between different media systems exceptions can occur due to file paths. They can happen between a local file system and a remote system based on blob storage. What is accepted on one system may be rejected on another as the file path is too long. Normally this will only happen for files with particularly long names.
+
+If you are happy to continue without throwing exceptions in these instances you can [modify the configuration](./deploy-setings#continuennmediafilepathtoolongexception). If this is done such files will be skipped, and although the media item will exist there will be no associated file.
+
+## Schema files following upgrades
+
+When Umbraco schema items are created, a representation of them is saved to disk as a `.uda` file in the `/data/revision/` folder.  The representation is known as an artifact. It will be refreshed on further updates to represent the current state of the schema item.
+
+Following an upgrade, it's possible the contents of the file will no longer match what would be generated by the current version.  For example, if a property has been added to an artifact, this will be absent in the file. It would be added if the file was recreated following the upgrade.
+
+This can lead to situations where Deploy continues to process a file it considers changed, even though the item represented is up-to-date. Which in turn means slow updates of Umbraco schema, as Deploy is processing more files than it needs to do.
+
+To resolve this situation, following an upgrade it is good practice to resave the `.uda` files in the "left-most" environment.  This will usually be the local one, or if not using that, the Development environment.  You can do this via the *Export Schema To Data Files* operation available on the _Settings > Deploy_ dashboard:
```

Comment:
```suggestion
When Umbraco schema items are created, a representation of them is saved to disk as a `.uda` file in the `/data/revision/` folder.  The representation is known as an artifact. It will be refreshed on further updates to represent the current state of the schema item.

Following an upgrade, it's possible the contents of the file will no longer match what would be generated by the current version.  For example, if a property has been added to an artifact, this will be absent in the file. It would be added if the file was recreated following the upgrade.

This can lead to situations where Deploy continues to process a file it considers changed, even though the item represented is up-to-date. This in turn means slow updates of Umbraco schema, as Deploy is processing more files than it needs to do.

To resolve this situation, following an upgrade it is good practice to resave the `.uda` files in the "left-most" environment.  This will usually be the local one, or if not using that, the Development environment.  You can do this via the *Export Schema To Data Files* operation available on the _Settings > Deploy_ dashboard:
```